main.go

```diff
@@ -38,7 +144,175 @@ func httpHandler(w http.ResponseWriter, r *http.Request) {
 			" Please check back again later.")); err != nil {
 			log.Printf("error during Write: %v", err)
 		}
+	case http.MethodPost:
+		http.HandleFunc("/verifyProject", verifySignature)
 	default:
 		http.Error(w, "only GET method is allowed", http.StatusMethodNotAllowed)
 	}
 }
+
+func verifyScorecardWorkflow(workflowContent string) bool {
+	// Verify workflow contents using actionlint.
+	workflow, lintErrs := actionlint.Parse([]byte(workflowContent))
+	if lintErrs != nil {
+		return false
+	}
+
+	// Extract main job
+	jobs := workflow.Jobs
+	if len(jobs) != 1 {
+		return false
+	}
+	analysisJob := jobs["analysis"]
+	if analysisJob == nil {
+		return false
+	}
+
+	// Verify that there is no container or services.
+	if analysisJob.Container != nil || len(analysisJob.Services) > 0 {
+		return false
+	}
+
+	// Verify that the workflow runs on ubuntu-latest.
+	if len(analysisJob.RunsOn.Labels) != 1 || analysisJob.RunsOn.Labels[0].Value != "ubuntu-latest" {
```

Comment:
this will crash if `runs-on` is not set in the workflow? 
Also, if `len(analysisJob.RunsOn.Labels) != 0`, the left part of the statement will run and crash because of indexing `Label[0]`

Reply:
Fixed. Its a bit messy now but idk if there is a cleaner way to check for this